From afc665263939db754b52ee2f00134419f9c316a9 Mon Sep 17 00:00:00 2001
From: dzhang <daniel.zhang@aviatnet.com>
Date: Tue, 12 Jan 2021 17:05:27 +1300
Subject: [PATCH] Fix uart_asm_init and tzpc_asm_init

---
 board/samsung/smdkc100/lowlevel_init.S | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/board/samsung/smdkc100/lowlevel_init.S b/board/samsung/smdkc100/lowlevel_init.S
index c55a4d3..60057c0 100644
--- a/board/samsung/smdkc100/lowlevel_init.S
+++ b/board/samsung/smdkc100/lowlevel_init.S
@@ -302,6 +302,7 @@ system_ddr_init:
  * uart_asm_init: Initialize UART's pins
  */
 uart_asm_init:
+	ldr	r8, =S5PC100_GPIO_BASE
 	mov	r0, r8
 	ldr	r1, =0x22222222
 	str	r1, [r0, #0x0]			@ GPA0_CON
@@ -314,20 +315,26 @@ uart_asm_init:
  * tzpc_asm_init: Initialize TZPC
  */
 tzpc_asm_init:
-	ldr	r0, =0xE3800000
+	ldr	r0, =0xF1500000
 	mov	r1, #0x0
 	str	r1, [r0]
 	mov	r1, #0xff
 	str	r1, [r0, #0x804]
 	str	r1, [r0, #0x810]
+	str	r1, [r0, #0x81C]
 
-	ldr	r0, =0xE2800000
+	ldr	r0, =0xFAD00000
 	str	r1, [r0, #0x804]
 	str	r1, [r0, #0x810]
 	str	r1, [r0, #0x81C]
 
-	ldr	r0, =0xE2900000
+	ldr	r0, =0xE0600000
 	str	r1, [r0, #0x804]
 	str	r1, [r0, #0x810]
+	str	r1, [r0, #0x81C]
+	str	r1, [r0, #0x828]
+	
+	ldr	r0, =0xE1C00000
+	str	r1, [r0, #0x804]
 
 	mov	pc, lr
-- 
2.7.4

