From f6068b75920746db5dac2a8b5a3ddb77e42c955b Mon Sep 17 00:00:00 2001
From: dzhang <daniel.zhang@aviatnet.com>
Date: Tue, 12 Jan 2021 16:42:10 +1300
Subject: [PATCH] Add s5pv210 DDR init

---
 board/samsung/smdkc100/lowlevel_init.S | 182 +++++++++++++++++++++++++++++++++
 1 file changed, 182 insertions(+)

diff --git a/board/samsung/smdkc100/lowlevel_init.S b/board/samsung/smdkc100/lowlevel_init.S
index fbfd9ab..c55a4d3 100644
--- a/board/samsung/smdkc100/lowlevel_init.S
+++ b/board/samsung/smdkc100/lowlevel_init.S
@@ -57,6 +57,9 @@ lowlevel_init:
 	
 	/* for CLOCK */
 	bl system_clock_init
+	
+	/* for DDR */
+	bl system_ddr_init
 
 	/* for UART */
 	bl uart_asm_init
@@ -117,6 +120,185 @@ system_clock_init:
 	mov	pc, lr
 
 /*
+ * system_ddr_init: Initialize DDR memory
+ */
+system_ddr_init:
+	ldr	r8, =S5PC100_DMC0_BASE		@ 0xF0000000
+
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101000
+	str	r1, [r8, #0x18]
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101002
+	str	r1, [r8, #0x18]
+	/* PHYCONTROL1 */
+	ldr	r1, =0x00000086
+	str	r1, [r8, #0x1C]
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101003
+	str	r1, [r8, #0x18]
+
+	/* wait at least 200us to stablize all clock */
+	mov	r2, #0x100000
+1:	subs	r2, r2, #1
+	bne	1b
+	
+	/* CONCONTROL */
+	ldr	r1, =0x0FFF2350
+	str	r1, [r8, #0x0]
+	/* MEMCONTROL */
+	ldr	r1, =0x00202430
+	str	r1, [r8, #0x4]
+	/* MEMCONFIG0 */
+	ldr	r1, =0x20E01323
+	str	r1, [r8, #0x8]
+	
+	/* prechconfig */
+	ldr	r1, =0xFF000000
+	str	r1, [r8, #0x14]
+	/* pwrdnconfig */
+	ldr	r1, =0xFFFF00FF
+	str	r1, [r8, #0x28]
+	
+	/* timingaref */
+	ldr	r1, =0x00000618
+	str	r1, [r8, #0x30]
+	/* timingrow */
+	ldr	r1, =0x19233309
+	str	r1, [r8, #0x34]
+	/* timingdata */
+	ldr	r1, =0x23240204
+	str	r1, [r8, #0x38]
+	/* timingpower */
+	ldr	r1, =0x09C80232
+	str	r1, [r8, #0x3c]
+	
+	/* directcmd */
+	ldr	r1, =0x07000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x01000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00020000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00030000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010400
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00000542
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x01000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x05000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x05000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00000442
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010780
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010400
+	str	r1, [r8, #0x10]
+	
+	/* CONCONTROL */
+	ldr	r1, =0x0FF02030
+	str	r1, [r8, #0x0]
+	/* pwrdnconfig */
+	ldr	r1, =0xFFFF00FF
+	str	r1, [r8, #0x28]
+	/* MEMCONTROL */
+	ldr	r1, =0x00202400
+	str	r1, [r8, #0x4]
+	
+	/* DMC1 */
+	ldr	r8, =S5PC100_DMC1_BASE		@ 0xF1400000
+	
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101000
+	str	r1, [r8, #0x18]
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101002
+	str	r1, [r8, #0x18]
+	/* PHYCONTROL1 */
+	ldr	r1, =0x00000086
+	str	r1, [r8, #0x1C]
+	/* PHYCONTROL0 */
+	ldr	r1, =0x00101003
+	str	r1, [r8, #0x18]
+	
+/* wait at least 200us to stablize all clock */
+	mov	r2, #0x100000
+1:	subs	r2, r2, #1
+	bne	1b
+	
+	/* CONCONTROL */
+	ldr	r1, =0x0FFF2350
+	str	r1, [r8, #0x0]
+	/* MEMCONTROL */
+	ldr	r1, =0x00202430
+	str	r1, [r8, #0x4]
+	/* MEMCONFIG0 */
+	ldr	r1, =0x20E01323
+	str	r1, [r8, #0x8]
+	
+	/* prechconfig */
+	ldr	r1, =0xFF000000
+	str	r1, [r8, #0x14]
+	/* pwrdnconfig */
+	ldr	r1, =0xFFFF00FF
+	str	r1, [r8, #0x28]
+	
+	/* timingaref */
+	ldr	r1, =0x00000618
+	str	r1, [r8, #0x30]
+	/* timingrow */
+	ldr	r1, =0x19233309
+	str	r1, [r8, #0x34]
+	/* timingdata */
+	ldr	r1, =0x23240204
+	str	r1, [r8, #0x38]
+	/* timingpower */
+	ldr	r1, =0x09C80232
+	str	r1, [r8, #0x3c]
+	
+	/* directcmd */
+	ldr	r1, =0x07000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x01000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00020000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00030000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010400
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00000542
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x01000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x05000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x05000000
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00000442
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010780
+	str	r1, [r8, #0x10]
+	ldr	r1, =0x00010400
+	str	r1, [r8, #0x10]
+	
+	/* CONCONTROL */
+	ldr	r1, =0x0FF02030
+	str	r1, [r8, #0x0]
+	/* pwrdnconfig */
+	ldr	r1, =0xFFFF00FF
+	str	r1, [r8, #0x28]
+	/* MEMCONTROL */
+	ldr	r1, =0x00202400
+	str	r1, [r8, #0x4]
+
+	mov	pc, lr
+	
+/*
  * uart_asm_init: Initialize UART's pins
  */
 uart_asm_init:
-- 
2.7.4

