From 5f241b4bbf1ae86da473f0650c651fa93ca3534d Mon Sep 17 00:00:00 2001
From: dzhang <daniel.zhang@aviatnet.com>
Date: Tue, 12 Jan 2021 17:22:33 +1300
Subject: [PATCH] Fix disable all interrupts

---
 board/samsung/smdkc100/lowlevel_init.S | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/board/samsung/smdkc100/lowlevel_init.S b/board/samsung/smdkc100/lowlevel_init.S
index 60057c0..b6ea405 100644
--- a/board/samsung/smdkc100/lowlevel_init.S
+++ b/board/samsung/smdkc100/lowlevel_init.S
@@ -38,22 +38,26 @@ lowlevel_init:
 	ldr	r0, =S5PC100_VIC0_BASE			@0xE4000000
 	ldr	r1, =S5PC100_VIC1_BASE			@0xE4000000
 	ldr	r2, =S5PC100_VIC2_BASE			@0xE4000000
+	ldr	r3, =S5PC100_VIC3_BASE			@0xE4000000
 
 	/* Disable all interrupts (VIC0, VIC1 and VIC2) */
-	mvn	r3, #0x0
-	str	r3, [r0, #0x14]				@INTENCLEAR
-	str	r3, [r1, #0x14]				@INTENCLEAR
-	str	r3, [r2, #0x14]				@INTENCLEAR
+	mvn	r4, #0x0
+	str	r4, [r0, #0x14]				@INTENCLEAR
+	str	r4, [r1, #0x14]				@INTENCLEAR
+	str	r4, [r2, #0x14]				@INTENCLEAR
+	str	r4, [r3, #0x14]				@INTENCLEAR
 
 	/* Set all interrupts as IRQ */
 	str	r5, [r0, #0xc]				@INTSELECT
 	str	r5, [r1, #0xc]				@INTSELECT
 	str	r5, [r2, #0xc]				@INTSELECT
+	str	r5, [r3, #0xc]				@INTSELECT
 
 	/* Pending Interrupt Clear */
 	str	r5, [r0, #0xf00]			@INTADDRESS
 	str	r5, [r1, #0xf00]			@INTADDRESS
 	str	r5, [r2, #0xf00]			@INTADDRESS
+	str	r5, [r3, #0xf00]			@INTADDRESS
 	
 	/* for CLOCK */
 	bl system_clock_init
@@ -81,9 +85,9 @@ system_clock_init:
 	/* Set Lock Time */
 	ldr	r1, =0xe10			@ Locktime : 0xe10 = 3600
 	str	r1, [r8, #0x000]		@ APLL_LOCK
-	str	r1, [r8, #0x004]		@ MPLL_LOCK
-	str	r1, [r8, #0x008]		@ EPLL_LOCK
-	str	r1, [r8, #0x00C]		@ HPLL_LOCK
+	str	r1, [r8, #0x008]		@ MPLL_LOCK
+	str	r1, [r8, #0x010]		@ EPLL_LOCK
+	str	r1, [r8, #0x020]		@ VPLL_LOCK
 
 	/* APLL_CON */
 	ldr	r1, =0x807d0301		@ SDIV 0, PDIV 4, MDIV 444 (1332MHz)
-- 
2.7.4

